### DISCUSSION TITLE
Are the reference swatch colours in the example notebook linear?

### DISCUSSION THREAD

**Discussion by tobias-kirschstein** [colour-science/colour-checker-detection#59] [General]
### Question

In the [example notebook](https://github.com/colour-science/colour-checker-detection/blob/master/colour_checker_detection/examples/examples_detection.ipynb), the images are transformed to linear RGB via `colour.cctf_decoding()`. This means that the colors of the detected `SWATCHES` are also given in linear RGB coordinates, right?
However, the `REFERENCE_SWATCHES` are loaded as `sRGB` color values and, as far as I am concerned, are not further converted to linear RGB. In the subsequent color correction 
`colour.colour_correction(COLOUR_CHECKER_IMAGES[i], swatches, REFERENCE_SWATCHES)` the original image is transformed such that the detected `swatches` match the `REFERENCE_SWATCHES`, but isn't this nonsensical as the values are given wrt different color spaces (linear RGB vs sRGB)? Or am I missing something?

Thanks.

**Comment by KelSolaar** [comment]
Hi @tobias-kirschstein,

The following code from the notebook and specifically the `colour.XYZ_to_RGB` [definition](https://colour.readthedocs.io/en/develop/generated/colour.XYZ_to_RGB.html):

```python
REFERENCE_SWATCHES = colour.XYZ_to_RGB(
        colour.xyY_to_XYZ(list(REFERENCE_COLOUR_CHECKER.data.values())),
        REFERENCE_COLOUR_CHECKER.illuminant, D65,
        colour.RGB_COLOURSPACES['sRGB'].matrix_XYZ_to_RGB)
```

produce linear values, thus `REFERENCE_SWATCHES` is effectively linear.

Cheers,

Thomas

**Comment by tobias-kirschstein** [comment]
Hi thanks for the quick reply.

Why does `colour.XYZ_to_RGB()` produce linear RGB values when used with `colour.RGB_COLOURSPACES['sRGB'].matrix_XYZ_to_RGB` from the `sRGB` color space? I though `sRGB` is non-linear?

Best,
Tobias

**Reply by KelSolaar** [reply]
Hi @tobias-kirschstein,

Some relevant reading: https://www.colour-science.org/posts/the-importance-of-terminology-and-srgb-uncertainty/

Cheers,

Thomas

**Reply by tobias-kirschstein** [reply]
Thanks for the link, that is a good reference.
So basically, `colour.XYZ_to_RGB()` always maps to an additive (=linear) RGB color space per default. 
To get the "normal" `sRGB` color space one can additionally set `cctf_encoding="sRGB"`.  
Essentially this means that there is a linear counterpart for every RGB colorspace, right?
I think some clarifying comments in the notebook could have resolved my confusion in the beginning.

Best,
Tobias

**Reply by KelSolaar** [reply]
> To get the "normal" sRGB color space one can additionally set cctf_encoding="sRGB".

You would need to pass a function explicitly, a [callable](https://docs.python.org/3.8/library/typing.html#typing.Callable) is expected:

![image](https://user-images.githubusercontent.com/99779/179375349-e0db5c04-304c-4cb8-aed5-a4b1c83f0d71.png)

> Essentially this means that there is a linear counterpart for every RGB colorspace, right?

Correct!

> I think some clarifying comments in the notebook could have resolved my confusion in the beginning.

I added a note here: ![image](https://user-images.githubusercontent.com/99779/179375501-7824f21e-51db-47be-afc6-ad41e540c133.png)

Cheers,

Thomas

**Reply by tobias-kirschstein** [reply]
Awesome! Thanks for your time this was very helpful!
